src/core/meshcop/dataset_updater.cpp

```diff
@@ -48,163 +48,164 @@ namespace MeshCoP {
 
 DatasetUpdater::DatasetUpdater(Instance &aInstance)
     : InstanceLocator(aInstance)
-    , mTimer(aInstance)
     , mDataset(nullptr)
 {
 }
 
 Error DatasetUpdater::RequestUpdate(const Dataset::Info &aDataset, UpdaterCallback aCallback, void *aContext)
 {
-    Error    error   = kErrorNone;
-    Message *message = nullptr;
+    Dataset dataset;
 
-    VerifyOrExit(!Get<Mle::Mle>().IsDisabled(), error = kErrorInvalidState);
-    VerifyOrExit(mDataset == nullptr, error = kErrorBusy);
+    dataset.SetFrom(aDataset);
+    return RequestUpdate(dataset, aCallback, aContext);
+}
+
+Error DatasetUpdater::RequestUpdate(Dataset &aDataset, UpdaterCallback aCallback, void *aContext)
+{
+    Error     error;
+    Message  *message = nullptr;
+    Dataset   activeDataset;
+    Timestamp activeTimestamp;
+    Timestamp pendingTimestamp;
+
+    error = kErrorInvalidState;
+    VerifyOrExit(!Get<Mle::Mle>().IsDisabled());
```

Comment:
Should this be `IsAttached()`?

Reply:
I think the original idea behind `IsDisabled()` check was to allow updating the dataset even while the device is attaching and currently detache. We could restrict this to `IsAttached()`, but I believe using `IsDisabled()` should work fine. What are your thoughts?

Reply:
Above, you stated:
> Restricts `DatasetUpdater::RequestUpdate()` to be used only when the device is fully attached and has a valid Active Dataset

What is this referring to?

Reply:
My intention was to ensure it doesn't have a partial dataset. It should either have the complete dataset (e.g., set by the next layer) or have successfully attached and received one.

Perhaps it's better to rephrase this and remove the reference to "fully attached."

Reply:
Updated in new push. The docs now say:

```c++
     * @retval kErrorInvalidState   Device is disabled or not fully configured (missing or incomplete Active Dataset).
```